src/sentry/workflow_engine/models/data_source.py

```diff
@@ -49,3 +52,19 @@ def type_handler(self) -> builtins.type[DataSourceTypeHandler]:
         if not handler:
             raise ValueError(f"Unknown data source type: {self.type}")
         return handler
+
+
+@receiver(pre_save, sender=DataSource)
+def ensure_valid_type(sender, instance: DataSource, **kwargs):
```

Comment:
nit: this isn't really checking the validity, just checking that it exists and is registered

Reply:
hmm - the way i was thinking of this is that we're validating the type on save, the validation constraints is just that it has a handler registered. Any suggestions on how to improve the naming? `ensure_type_handler_registered`?

Reply:
I like `ensure_type_handler_registered`!